idxmgmt/manager.go

```diff
@@ -134,8 +135,17 @@ func (m *manager) templateFeature(loadMode libidxmgmt.LoadMode) feature {
 }
 
 func (m *manager) ilmFeature(loadMode libidxmgmt.LoadMode) feature {
-	if m.supporter.st.ilmEnabled.Load() {
-		f := newFeature(true, false, true, loadMode)
+	// Do not use configured `m.supporter.ilmConfig.Mode` to check if ilm is enabled.
+	// The configuration might be set to `true` or `auto` but preconditions are not met,
+	// e.g. ilm support by Elasticsearch
+	// In these cases the supporter holds an internal state `m.supporter.st.ilmEnabled` that is set to false.
+	// The originally configured value is preserved allowing to collect warnings and errors to be
+	// returned to the user.
+
+	// m.supporter.st.ilmEnabled.Load() only returns true for cases where
+	// ilm mode is configured `auto` or `true` and preconditions to enable ilm are true
+	if enabled := m.supporter.st.ilmEnabled.Load(); enabled {
```

Comment:
if this returns a bool, shouldn't be named `Loaded()` or `IsLoaded()`?

Reply:
`m.supporter.st.ilmEnabled` is an [atomic.Bool](https://github.com/uber-go/atomic/blob/master/atomic.go#L232).